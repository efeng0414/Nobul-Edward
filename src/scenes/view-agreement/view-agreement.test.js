import React from "react";
import renderer from "react-test-renderer";
import ViewAgreementWrapped from "./container";
import { IntlProvider, intlShape } from "react-intl";

const { WrappedComponent: ViewAgreement } = ViewAgreementWrapped;

describe("View Agreement", () => {
  const mockGetAgreementData = jest.fn();
  const containerProps = {
    intl: global.getIntl(),
    getAgreementData: mockGetAgreementData,
    match: {
      params: {
        agreementType: "buy",
        agreementId: "123ab"
      }
    },
    agreement: {}
  };
  const containerJSX = <ViewAgreement {...containerProps} />;

  describe("Container Snapshot", () => {
    const snapshotJSX = <IntlProvider {...global.intl}>{containerJSX}</IntlProvider>;

    const containerSnapshot = renderer.create(snapshotJSX).toJSON();
    it("should match stored snapshot", () => {
      expect(containerSnapshot).toMatchSnapshot();
    });
  });

  describe("Container Mount", () => {
    it("should call getAgreementData in componentDidMount", () => {
      mockGetAgreementData.mockReset();
      const containerMount = global.Enzyme.shallow(
        containerJSX,
        global.getIntlContext({ intl: containerProps.intl })
      );
      expect(mockGetAgreementData).toBeCalledWith("buy", "123ab");
    });
  });
});
