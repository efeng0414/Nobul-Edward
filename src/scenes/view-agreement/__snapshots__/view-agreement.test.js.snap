// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Agreement Container Snapshot should match stored snapshot 1`] = `
<div>
  <div
    className="ant-layout"
  >
    <h2>
      buy
       agreement
    </h2>
    <div>
      <span>
        Client Name
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Brokerage Name
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Brokerage Address
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Agent Name
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Agent phone number
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Agent email address
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Start date
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Expiry Date
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Holdover period (days)
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        MLSÂ®#
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Property area
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Commission
      </span>
       -
      <span>
        ...
      </span>
    </div>
    <div>
      <span>
        Property Type
      </span>
       -
      <span>
        ...
      </span>
    </div>
  </div>
</div>
`;
